### DISCUSSION TITLE
It is possible to convert TSESTree to TypeScript AST?

### DISCUSSION THREAD

**Discussion by bjornharrtell** [typescript-eslint/typescript-eslint#7228] [Technical Discussions]
https://typescript-eslint.io/blog/asts-and-typescript-eslint/ says typescript-eslint does conversion of TypeScript AST to TSESTree but then also says it keeps them in sync, letting me to believe it might be possible to simply convert TSESTree to TypeScript AST?

The reason I'm interested in this is because I currently generate ESTree with my own custom tooling from Java, then generate JavaScript from that. I would like to find a migration path to generate TypeScript instead.

**Comment by bradzacher** [comment]
There's no reason you couldn't reverse the conversion - but you'd have to write that logic yourself. TS does provide the APIs to create ast nodes so it's possible, it'll be a decent chunk of work to say the least.

When we parse TS code we do so with TS itself, then we convert the TS AST to our TS-ESTree representation. We keep a mapping of the AST nodes though so we can move back and forth between the ASTs.

**Reply by bjornharrtell** [reply]
Thanks for the informative response!
